src/meta-srv/src/region/supervisor.rs

```diff
@@ -389,75 +422,121 @@ impl RegionSupervisor {
+                ensure!(
+                    peers.len() == regions.len(),
+                    error::PeerCountMismatchSnafu {
+                        expected: regions.len(),
+                        actual: peers.len(),
+                    }
+                );
+                let region_peers = regions
+                    .iter()
+                    .zip(peers)
+                    .map(|(region_id, peer)| (*region_id, peer))
+                    .collect::<Vec<_>>();
+
+                Ok(region_peers)
+            }
+            RegionSupervisorSelector::RegionStatAwareSelector(selector) => {
+                let peers = selector
+                    .select(&self.selector_context, from_peer_id, regions)
+                    .await?;
+
+                ensure!(
+                    peers.len() == regions.len(),
+                    error::PeerCountMismatchSnafu {
+                        expected: regions.len(),
+                        actual: peers.len(),
+                    }
+                );
+                Ok(peers)
+            }
+        }
+    }
+
+    async fn generate_failover_tasks(
+        &mut self,
+        from_peer_id: DatanodeId,
+        regions: &[RegionId],
+    ) -> Result<Vec<RegionMigrationProcedureTask>> {
+        let mut tasks = Vec::with_capacity(regions.len());
         let from_peer = self
             .peer_lookup
-            .datanode(datanode_id)
+            .datanode(from_peer_id)
             .await
             .context(error::LookupPeerSnafu {
-                peer_id: datanode_id,
+                peer_id: from_peer_id,
             })?
             .context(error::PeerUnavailableSnafu {
-                peer_id: datanode_id,
+                peer_id: from_peer_id,
             })?;
-        let mut peers = self
-            .selector
-            .select(
-                &self.selector_context,
-                SelectorOptions {
-                    min_required_items: 1,
-                    allow_duplication: false,
-                    exclude_peer_ids: HashSet::from([from_peer.id]),
-                },
-            )
-            .await?;
-        let to_peer = peers.remove(0);
-        if to_peer.id == from_peer.id {
-            warn!(
-                "Skip failover for region: {region_id}, from_peer: {from_peer}, trying to failover to the same peer."
-            );
-            return Ok(());
+        let region_peers = self.select_peers(from_peer_id, regions).await?;
+
+        for (region_id, peer) in region_peers {
+            let count = *self
+                .failover_counts
+                .entry((from_peer_id, region_id))
+                .and_modify(|count| *count += 1)
+                .or_insert(1);
+            let task = RegionMigrationProcedureTask {
+                region_id,
+                from_peer: from_peer.clone(),
+                to_peer: peer,
+                timeout: DEFAULT_REGION_MIGRATION_TIMEOUT * count,
```

Comment:
Add log for the tries count

Reply:
Resolved in [5cc7d5d](https://github.com/GreptimeTeam/greptimedb/pull/5990/commits/5cc7d5d18ad1d69199e2a6d2d67a347b11373529)